frontend/src/pages/modelRegistrySettings/CreateModal.tsx

```diff
@@ -18,12 +23,30 @@ import NameDescriptionField from '~/concepts/k8s/NameDescriptionField';
 import { NameDescType } from '~/pages/projects/types';
 import FormSection from '~/components/pf-overrides/FormSection';
 import { AreaContext } from '~/concepts/areas/AreaContext';
+import { SupportedArea, useIsAreaAvailable } from '~/concepts/areas';
+import SearchSelector from '~/components/searchSelector/SearchSelector';
+import { PemFileUpload } from './PemFileUpload';
 
 type CreateModalProps = {
   onClose: () => void;
   refresh: () => Promise<unknown>;
 };
 
+enum SecureDBRadios {
+  CLUSTER_WIDE = 'cluster-wide',
+  OPENSHIFT = 'openshift',
+  EXISTING = 'existing',
+  NEW = 'new',
+}
+
+type SecureDBInfo = {
+  radio: SecureDBRadios;
+  nameSpace: string;
+  configMap: string;
+  certificate: string;
+  key: string;
+};
+
```

Comment:
Wondering if these types should be here or in some `types` file? Maybe `frontend/src/concepts/modelRegistry/types.ts`?

Reply:
@manaswinidas - honestly i didn't want to do something too permanent as i'm sort of guessing on what some of the data will look like. also this (especially the `radio` part is very specific to this form and how it shows/hides/validates different things)

Reply:
Yeah, we may want to consider reorganizing some of these types as part of API submit logic in [RHOAIENG-15900](https://issues.redhat.com/browse/RHOAIENG-15900) (cc @ppadti) but I don't think it needs to block this PR